StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -181,82 +182,85 @@ extension PaymentSheet {
 
         // MARK: - New Payment Method
         case let .new(confirmParams):
-            let paymentMethodType: STPPaymentMethodType = {
-                switch paymentOption.paymentMethodType {
-                case .stripe(let paymentMethodType):
-                    return paymentMethodType
-                default:
-                    return .unknown
-                }
-            }()
-            // Set allow_redisplay on params
-            confirmParams.setAllowRedisplay(
-                mobilePaymentElementFeatures: elementsSession.customerSessionMobilePaymentElementFeatures,
-                isSettingUp: intent.isSetupFutureUsageSet(for: paymentMethodType)
-            )
-            confirmParams.setClientAttributionMetadata(clientAttributionMetadata: clientAttributionMetadata)
-            switch intent {
-            // MARK: ↪ PaymentIntent
-            case .paymentIntent(let paymentIntent):
-                let params = makePaymentIntentParams(
-                    confirmPaymentMethodType: .new(
-                        params: confirmParams.paymentMethodParams,
-                        paymentOptions: confirmParams.confirmPaymentMethodOptions,
-                        shouldSave: confirmParams.saveForFutureUseCheckboxState == .selected,
-                        shouldSetAsDefaultPM: confirmParams.setAsDefaultPM
-                    ),
-                    paymentIntent: paymentIntent,
-                    configuration: configuration
-                )
-                paymentHandler.confirmPayment(
-                    params,
-                    with: authenticationContext,
-                    completion: { actionStatus, paymentIntent, error in
-                        if let paymentIntent {
-                            setDefaultPaymentMethodIfNecessary(actionStatus: actionStatus, intent: .paymentIntent(paymentIntent), configuration: configuration, paymentMethodSetAsDefault: elementsSession.paymentMethodSetAsDefaultForPaymentSheet)
-                        }
-                        paymentHandlerCompletion(actionStatus, error)
+            PassiveCaptcha.fetchPassiveHCaptchaToken(passiveCaptcha: elementsSession.passiveCaptcha) { hcaptchaToken in
```

Comment:
Can we fetch the passive HCaptcha token earlier? Like maybe when PaymentSheet initializes, and then we could have it ready when someone actually hits "Pay"?